proto/tendermint/abci/types.proto

```diff
@@ -185,17 +185,23 @@ message ResponseBeginBlock {
   repeated Event events = 1 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
 }
 
+message MempoolError {
+  string error_message = 1;
+  bool   is_retryable  = 2;
+}
+
 message ResponseCheckTx {
-  uint32         code       = 1;
-  bytes          data       = 2;
-  string         log        = 3;  // nondeterministic
-  string         info       = 4;  // nondeterministic
-  int64          gas_wanted = 5 [json_name = "gas_wanted"];
-  int64          gas_used   = 6 [json_name = "gas_used"];
-  repeated Event events     = 7 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
-  string         codespace  = 8;
-  string         sender     = 9;
-  int64          priority   = 10;
+  uint32         code          = 1;
+  bytes          data          = 2;
+  string         log           = 3;  // nondeterministic
+  string         info          = 4;  // nondeterministic
+  int64          gas_wanted    = 5 [json_name = "gas_wanted"];
+  int64          gas_used      = 6 [json_name = "gas_used"];
+  repeated Event events        = 7 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
+  string         codespace     = 8;
+  string         sender        = 9;
+  int64          priority      = 10;
+  MempoolError   mempool_error = 11;
```

Comment:
```suggestion
  MempoolError   mempool_error = 11;
```

What are your thoughts on having this just be an `error`, i.e. a `string`?

Reply:
I think it depends on how we expect the client to handle the error. If we expect the client to just log the error and do nothing, then I think it makes sense to have the error be just a string. If the client should react to the error in a programmatic way, I think we should have metadata that helps the client decide what to do about the error. 

Do you have any sense for how the client will use this error?

Reply:
> I think it depends on how we expect the client to handle the error. If we expect the client to just log the error and do nothing, then I think it makes sense to have the error be just a string.

This is exactly how it'll happen. The response will come back to the client and it'll be printed in the CLI or returned as an API response. The client will see "something about gas" and will either try again or move on.

The client will not react in a programmatic way to this AFAIK.

Reply:
Great. Then no need to fuss over that. I'll move that field to a string, put up tests for this change and move it out of draft.